src/chart/generator/marker.h

```diff
@@ -43,11 +43,12 @@ class Marker
 
 	struct Label
 	{
+		constexpr static uint64_t INVALID_MEASURE = static_cast<uint64_t>(-1);
 		double value;
 		Data::ColumnIndex measureId;
 		std::string unit;
 		std::string indexStr;
-		Label() : value(0.0), measureId(-1) {}
+		Label() : value(0.0), measureId(INVALID_MEASURE) {}
```

Comment:
we should have used std::optional here in the first place

Reply:
It can rewrite to optional. I handle it in a separated PR

Reply:
ok, cool